src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -207,7 +205,10 @@ function getAdaptationSetSwitchingIDs(
 }
 
 /**
- * Process intermediate periods to create final parsed periods.
+ * Process AdaptationSets intermediate representations to return under its final
+ * form.
+ * Note that the AdaptationSets returned are sorted by priority (from the most
```

Comment:
Why not deciding to sort Adaptations when needed (as before applying preferences) ?
Other parts of code are not supposed to know that it is ordered.

Reply:
You mean adding the `selectionPriority` property to an `Adaptation` object and using it?

I find the already-ordered solution more elegant. It makes sense to me that parsed Adaptations are sorted by priority.